### PR TITLE
-> longest_repeated_subsequence

### PR DESCRIPTION
**Pull request by smichr**
add functions to iterables

### CODE REVIEW

**Comment by asmeurer** [comment]
Can you give an example for the first case?

File: sympy/simplify/tests/test_cse.py [hunk 1]
```diff
@@ -185,17 +187,19 @@ def test_noncommutative_mul_cse():
     c = E*B*C*D*E
 
     repls, exprs = cse([a, b, c])
-    assert repls == [(x0, E*B), (x1, A*x0*A), (x2, D*E), (x3, C*x2)]
-    assert exprs == [x1*x3, x1*x2, x0*x3]
+    assert repls == [(x0, A*E*B*A), (x1, D*E), (x2, C*x1)]
+    assert exprs == [x0*x2, x0*x1, E*B*x2]
```
Comment (asmeurer) [inline sympy/simplify/tests/test_cse.py]: This has fewer assignments, but more operations (5 in `repls` and 4 in `exprs` vs. 5 in `repls` and 3 in `exprs`). So I think the way it was before is better. The whole point of cse is to minimize the total number of operations at the cost of assigning more variables.

File: sympy/simplify/tests/test_cse.py [hunk 2]
```diff
@@ -498,19 +498,14 @@ def _replace_cse(strs, replacement_dict):
         return strs
 
     repl, strs = _cse_str([a, b, c])
-    assert (repl, strs) == ({'0': 'de', '1': 'c0', '2': 'ab', '3': 'a2a'}, ['31', '30', '21'])
+    assert (repl, strs) == ({'2': 'c1', '1': 'de', '0': 'aaba'}, ['02', '01', 'ab2'])
```
Comment (asmeurer) [inline sympy/simplify/tests/test_cse.py]: Same here as above. If these strings were multiplications, there would be more operations.

**Comment by asmeurer** [comment]
Given my line comments, I am -1 to this, since it seems to perform worse in terms of minimizing operations. [I mentioned on the original PR](https://github.com/sympy/sympy/pull/10271#issuecomment-166023202) that just always choosing common subsequences of length 2 always minimizes operations, but can produce unnecessary assignments (which could likely always be removed by checking which variables are only used once).  

I like the moving of `replace_subsequence` to iterables. That should be a separate commit.

**Comment by smichr** [comment]
I make comments in the parent PR.

**Comment by smichr** [comment]
> Can you give an example [of a problem with the main PR code]

I left a comment on the main PR page in the test code.

**Comment by smichr** [comment]
I will reopen the changes to iterables as a separate PR.

**Pull request closed without merging**